src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -488,6 +488,30 @@ inline void ShenandoahHeap::mark_complete_marking_context() {
   _marking_context->mark_complete();
 }
 
+inline void ShenandoahHeap::mark_finished() {
+  mark_complete_marking_context();
+
+  class ShenandoahCollectLiveSizeClosure : public ShenandoahHeapRegionClosure {
```

Comment:
We don't usually use the in-method declarations like these, pull it out of the method.

Reply:
👍